### PR TITLE
Fixes #5018 and enables last MP REST TCK test

### LINKED ISSUE #5018: ScopeInfo should have a public constructor
Otherwise it can't be used for custom scopes in a `BeanConfigurator`. We should also add some javadoc to instruct the users to use `io.quarkus.arc.processor.BuiltinScope` for built-in scopes.

### PR DESCRIPTION
**Pull request by kenfinnigan**

### CODE REVIEW

File: extensions/rest-client/deployment/src/main/java/io/quarkus/restclient/deployment/RestClientProcessor.java [hunk 1]
```diff
@@ -258,33 +261,46 @@ private String computeConfigPrefix(String interfaceName, ClassInfo classInfo) {
...
+
         if (scopeConfig.isPresent()) {
             final DotName scope = DotName.createSimple(scopeConfig.get());
             final BuiltinScope builtinScope = BuiltinScope.from(scope);
             if (builtinScope != null) { // override default @Dependent scope with user defined one.
-                scopeInfo = builtinScope.getInfo();
-            } else {
+                scopeToUse = builtinScope.getInfo();
+            } else if (capabilities.isCapabilityPresent(Capabilities.SERVLET)) {
+                if (scope.toString().equals("javax.enterprise.context.SessionScoped")) {
```
Comment (mkouba) [inline extensions/rest-client/deployment/src/main/java/io/quarkus/restclient/deployment/RestClientProcessor.java]: You should probably use some `DotName` constant or `SessionScoped.class.getName()`?

**Review by mkouba — CHANGES_REQUESTED** [review]
Otherwise looks good...

File: extensions/rest-client/deployment/src/main/java/io/quarkus/restclient/deployment/RestClientProcessor.java (hunk 1 above)
Comment (kenfinnigan) [reply extensions/rest-client/deployment/src/main/java/io/quarkus/restclient/deployment/RestClientProcessor.java]: Ok, wasn't sure about that

Will fix

**Comment by mkouba** [comment]
One more thing - could we add a constructor to the `ScopeInfo` with a `DotName` param instead of `Class`? ;-)

**Comment by kenfinnigan** [comment]
And then use that for SessionScoped creation?

**Comment by kenfinnigan** [comment]
@mkouba in the constructor, how do we know if it's @Inherited from the DotName?

We need to retrieve the class from somewhere?

**Comment by kenfinnigan** [comment]
How's that?

**Review by mkouba — APPROVED** [review]
Looks good.

**Pull request merged**